src/components/auth/signUp/signUpForm1.component.tsx

```diff
@@ -26,174 +24,133 @@ export interface SignUpForm1Type {
 
   public state: State = {
-    firstName: '',
-    lastName: '',
-    date: '',
-    email: '',
-    password: '',
-    firstNameValid: false,
-    lastNameValid: false,
-    dateValid: false,
-    emailValid: false,
-    passwordValid: false,
+    firstName: undefined,
+    lastName: undefined,
+    date: undefined,
+    email: undefined,
+    password: undefined,
     termsAccepted: false,
   };
 
-  private onFirstNameValidationResult = (firstNameValid: boolean, firstName: string) => {
-    this.setState({ firstNameValid, firstName });
+  public componentDidUpdate(prevProps: SignUpForm1Props, prevState: State) {
+    const oldFormValid: boolean = this.isValid(prevState);
+    const newFormValid: boolean = this.isValid(this.state);
+
+    const becomeValid: boolean = !oldFormValid && newFormValid;
+    const becomeInvalid: boolean = oldFormValid && !newFormValid;
+
+    if (becomeValid) {
+      this.props.onFormValueChange(this.state);
+    } else if (becomeInvalid) {
+      this.props.onFormValueChange(undefined);
+    }
+  }
+
+  private onFirstNameInputTextChange = (firstName: string) => {
+    this.setState({ firstName });
   };
 
-  private onLastNameValidationResult = (lastNameValid: boolean, lastName: string) => {
-    this.setState({ lastNameValid, lastName });
+  private onLastNameValidationResult = (lastName: string) => {
+    this.setState({ lastName });
   };
 
-  private onDateValidationResult = (dateValid: boolean, date: string) => {
-    this.setState({ dateValid, date });
+  private onDateInputTextChange = (date: string) => {
+    this.setState({ date });
   };
 
-  private onEmailValidationResult = (emailValid: boolean, email: string) => {
-    this.setState({ emailValid, email });
+  private onEmailInputTextChange = (email: string) => {
+    this.setState({ email });
   };
 
-  private onPasswordValidationResult = (passwordValid: boolean, password: string) => {
-    this.setState({ passwordValid, password });
+  private onPasswordInputTextChange = (password: string) => {
+    this.setState({ password });
   };
 
   private onTermsAcceptChange = (termsAccepted: boolean) => {
     this.setState({ termsAccepted });
   };
 
-  private isFormValid = (): boolean => {
-    const {
-      firstNameValid,
-      lastNameValid,
-      dateValid,
-      emailValid,
-      passwordValid,
-      termsAccepted,
-    } = this.state;
-
-    return firstNameValid && lastNameValid && dateValid && emailValid && passwordValid && termsAccepted;
-  };
+  private isValid = (value: SignUpForm1Type): boolean => {
+    const { firstName, lastName, date, email, password, termsAccepted } = value;
 
-  private onSubmitPress = () => {
-    const { firstName, lastName, date, email, password } = this.state;
-
-    this.props.onSubmit({ firstName, lastName, date, email, password });
-  };
-
-  private getInputStatus = (valid: boolean): string => {
-    return valid ? 'success' : 'danger';
+    return firstName !== undefined
```

Comment:
```return !firstName && !whatever```

Reply:
Forms emit `change` event in same way like `ValidationInput`